blocks/library/embed/index.js

```diff
@@ -26,7 +27,33 @@ import BlockAlignmentToolbar from '../../block-alignment-toolbar';
 // These embeds do not work in sandboxes
 const HOSTS_NO_PREVIEWS = [ 'facebook.com' ];
 
-function getEmbedBlockSettings( { title, icon, category = 'embed', transforms, keywords = [] } ) {
+// Caches the embed API calls, so if blocks get transformed, or deleted and added again, we don't spam the API.
+const wpEmbedAPI = memoize( ( url ) => wp.apiRequest( { path: `/oembed/1.0/proxy?${ stringify( { url } ) }` } ) );
```

Comment:
Once #5219 lands, we're going to want to update this block to not depend on `wp.apiRequest` directly.

Reply:
I'd like to update this to use `withSelect` in a separate PR as it's a big change to the edit component, and I'd rather keep this PR as small as possible. Would this PR be ok to go in as it is? I've already started removing the direct `wp.apiRequest` call, but for ease of review I'd rather it was another PR.